lib/metrics/events.js

```diff
@@ -98,6 +98,7 @@ module.exports = log => {
         }
 
         return emitFlowEvent(event, request, data)
+          .then(() => {})
```

Comment:
To avoid making two calls to `request.gatherMetricsContext` below, I return the metrics context from `emitFlowEvent`. Added this so it doesn't leak any further.